api-lib/profileImageMutations.ts

```diff
@@ -0,0 +1,69 @@
+import { VercelRequest } from '@vercel/node';
+
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+  uploadImageInput,
+} from '../src/lib/zod';
+
+import { gql } from './Gql';
+
+export const userAndImageData = (
+  req: VercelRequest
+): {
+  input: { image_data_base64: string };
+  hasuraProfileId: number;
+} => {
+  const {
+    input: { object: input },
+    session_variables: sessionVariables,
+  } = composeHasuraActionRequestBodyWithSession(
+    uploadImageInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+  return { input, hasuraProfileId: sessionVariables.hasuraProfileId };
+};
+
+export const profileUpdateImageMutation = (
+  hasuraProfileId: number,
+  _set: (fileName: string) => any
+) => {
+  // save the new image id in the db
+  return async (fileName: string): Promise<any> => {
+    const mutationResult = await gql.q('mutation')({
+      update_profiles_by_pk: [
+        {
+          _set: _set(fileName),
```

Comment:
Is this needed to override some annoying type mismatch? If so, consider returning `unknown` here instead of `any` for the `_set` function. Also, it'd be great to leave a comment here explaining why this is needed since this effectively turns off the zeus type tooling.

Reply:
pushed a small change here, still dont love having to pass in a func